chips/nrf52/src/chip.rs

```diff
@@ -1,20 +1,18 @@
 use crate::deferred_call_tasks::DeferredCallTask;
-use crate::interrupt_service::InterruptService;
-use crate::nvmc;
 use core::fmt::Write;
 use cortexm4::{self, nvic};
 use kernel::common::deferred_call;
-use kernel::debug;
+use kernel::InterruptService;
 
-pub struct NRF52<I: InterruptService> {
+pub struct NRF52<I: InterruptService<DeferredCallTask> + 'static> {
```

Comment:
Why is a reference now needed? And if we're introducing new references, can we use a generic `'a` lifetime parameter, in the spirit of #1074?

Reply:
A reference is now needed because the struct that implements the interrupt service now also holds all the peripherals, and there are instances where the peripherals need to be instantiated before the Chip struct itself (maybe not for the nordic boards, but I think its good to be consistent across all chips). 

I changed it to use a generic lifetime.